### PR TITLE
Final: Updating this to support php8.x still supports 5.x and 7.x

### PR DESCRIPTION
**Pull request by fr3nch13**
This is a new pull request. I closed the other one. This one supports PHP 5.5, 5.6, 7.0, 7.1, 7.2, 7.4, 8.0, 8.1, 8.2.

It stems from an error I am getting with php 8.2:
```
deprecated: 8192 :: Creation of dynamic property josegonzalez\Dotenv\Expect::$environment is deprecated on line 14 of vendor/josegonzalez/dotenv/src/josegonzalez/Dotenv/Expect.php
```

This fixes that and updates the CI workflow.

### CODE REVIEW

**Comment by fr3nch13** [comment]
This doesn't seem to be including the github actions, but I think that's because your `master` branch doesn't have them setup in the first place.
However, you can see the latest actions from my fork here: https://github.com/302dev/php-dotenv/actions

It also uses docdecov.io coverage reports as seen in this pull request: https://github.com/302dev/php-dotenv/pull/1 and here https://app.codecov.io/gh/302dev/cakephp
It uses this github app: https://github.com/marketplace/codecov
Basically, you just need to configure them to talk to each other, then add the secret `CODECOV_TOKEN` to your github repo here: https://github.com/josegonzalez/php-dotenv/settings/secrets/actions

**Comment by fr3nch13** [comment]
Is there any way you could set up another branch that could be used for testing/dev before the code goes to `master`? Something like `version-next` or `development`? Then I could switch this pull request to point to that.
I ask, so that I can update my composer.json to use `josegonzales/php-dotenv:development` instead of `master` so the php 8.2.2 error will go away in my unit testing. I really don't want to point it to my fork.

**Comment by fr3nch13** [comment]
Closing as I need the pull request to use a different branch. See #55
